archive/install_shield_3_z.ksy

```diff
@@ -0,0 +1,484 @@
+          If the data is not compressed,
+          this is equal to len_data_compressed.
+      - id: len_data_compressed
+        type: u4
+        doc: |
+          Byte length of the data as stored in the archive,
+          i. e. possibly compressed.
+      - id: ofs_data
+        type: u4
+        doc: Absolute byte offset of the file data.
+      - id: modified
+        type: dos_datetime_backwards
+        doc: Modification date/time of the file.
+      - id: attributes
+        type: u4
+        doc: |
+          DOS file attributes of the file.
+
+          Only bits 0 through 5 are actually used by DOS/Windows -
+          all other bits should be zero.
+          Of these,
+          bits 3 and 4 mark volume labels and directories (respectively),
+          so they should never be set on files in an archive.
+
+          For some reason,
+          this field apparently always has bit 7 set if no other attributes are set.
+          This seems to have no meaning and can probably be ignored.
+      - id: len_entry
+        type: u2
+        doc: Total byte size of this file entry.
+      - id: reserved_flags_1
+        type: b4
+      - id: is_uncompressed
+        type: b1
+        doc: |
+          Whether this file's data is stored uncompressed.
+          If false,
+          the data is compressed using the PKWARE DCL Implode compression algorithm.
+      - id: internal_flag
+        type: b1
+        valid: false
+        doc: |
+          According to idecomp.py,
+          this flag is used internally by InstallShield's icomp
+          and should never be true in archive files.
+      - id: has_version
+        type: b1
+        doc: |
+          May be true if this file has a version number stored in the version field.
+          This flag isn't always set reliably
+          (see also the comments in idecomp.py) -
+          it's often false even for files which have a non-zero version number stored.
+      - id: reserved_flags_2
+        type: b1
+        valid: false
+      - id: is_split
+        type: b1
+        valid:
+          expr: _root.header.is_extended or not _
+        doc: |
+          Whether this file is split across multiple archive parts.
+          Only used in extended format -
+          otherwise this field is 0
+          and the file's data is not split.
+      - id: reserved_flags_3
+        type: b7
+        valid: 0
+      - id: reserved_1
+        contents: [0x00]
+        doc: |
+          Unused according to idecomp.py.
+          Seems to be always 0.
+      - id: start_part
+        type: u1
+        valid:
+          expr: |
+            is_split
+            ? (_ > 0 and _ < end_part)
+            : _ == end_part
+        doc: |
+          Number of the split archive part in which the first part of this file's data is stored.
+          Only used in extended format -
+          otherwise this field is 0
+          and the file's data is not split.
+      - id: len_name
+        type: u1
```

Comment:
`-affected-by: 84`

Reply:
Which part exactly? I'm not sure how any of the features in #84 would help with calculating "size of this type except for one of its fields"... Having `_sizeof` on variable-sized fields won't make a difference, because we already have an explicit `len_entry` field.

Reply:
>`sizeof(property_name1, property_name2)` is equivalent to
`lea(property_name2) - lea(property_name1) + sizeof(property_name2)`, which is meant to spare us from summing the fields lengths explicitly ourselves